lib/associations/belongs-to-many.js

```diff
@@ -298,6 +298,46 @@ var BelongsToMany = function(source, target, options) {
 
 util.inherits(BelongsToMany, Association);
 
+BelongsToMany.prototype.get = function(instances, options) {
+  var throughModel = this.throughModel;
+
+  var throughOptions = _.assign({
+    where: _.set({}, this.foreignKey, {$in: _.pluck(instances, this.source.primaryKeyAttribute)}),
+    logging: options.logging
+  });
+
+  var association = this;
+
+  return throughModel.findAll(throughOptions).then(function (throughs) {
+    var otherKey = association.otherKey
+      , targetIds = _.chain(throughs).pluck(otherKey).uniq().value();
+
+    var targetOptions = _.chain({
+      where: _.set({}, association.target.primaryKeyAttribute, {$in: targetIds})
+    })
+        .assign(association.target.$optClone(options))
+        .omit(['include'])
+        .value();
+
+    return Utils.Promise.join(throughs, _.isEmpty(targetIds) ? null : association.target.findAll(targetOptions));
+  }).then(_.spread(function(throughs, targets) {
+    targets = _.groupBy(targets, association.target.primaryKeyAttribute);
+    
+    return _.chain(instances)
+      .pluck(association.source.primaryKeyAttribute)
+      .map(function(sourceId) {
+        var instanceThroughs = _.filter(throughs, function (through) {
+          return through.get(association.foreignKey, {raw: true}) === sourceId;
+        })
+          , instanceTargets = _.at(targets, _.pluck(instanceThroughs, association.otherKey));
+
+        return [sourceId, _(instanceTargets).compact().flatten().value()];
+      })
+      .zipObject()
+      .value();
+  }));
+};
+
```

Comment:
Please refactor `injectGetter` to use this code now as done in the other include.seperate PR, otherwise we just have duplicate code.

Ideally you should also move the other methods like done in the other PR, but not stricly necessary.